### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by dumblob** [gurgeh/selfspy#178]
There seems to be lots of PRs waiting for merge, but nothing seems to happen for about 5 years already. Many of those PRs seem to be reviewed and tested, so it seems there are no technical reasons for no activity, but rather a lack of time.

@gurgeh could you grant admin rights to few other persons? I'm asking for admin so that they can also grant further rights to others.

**Comment by bob-ortiz** [comment]
The only member of https://github.com/selfspy is still a very active developer (https://anarc.at/). I'm not sure how this works exactly but maybe contacting him and try to get admin rights to other people. This project seems very powerful to let it die.

**Comment by dumblob** [comment]
Let's invite him to this thread. @anarcat what do you think?

**Comment by anarcat** [comment]
whoa, i'm a member here? totally forgot about that. i would not grant admin access to anyone before at least getting a :+1: from @gurgeh here. and even then, i tend to grant commit access to people that have provided good MRs in the past.

sorry. i am around though so if this falls through, we can figure something out.

**Comment by anarcat** [comment]
uh! so i was granted access when i asked for it in https://github.com/selfspy/selfspy/issues/161... so i guess i am a member of the project officially. totally forgot about that, which probably makes me a bad maintainer. not sure how to proceed going forward.

**Comment by grandchild** [comment]
i'm just a user and the arch package maintainer, but ideally this org would simply liberally accept PRs in order to make selfspy run on python 3 at least (the first candidate might probably be #174 from @giansegato ). there also seem to be some fairly active forks in https://github.com/selfspy/selfspy/network .

i'd be willing to do code review, especially for linux things. i cannot maintain this.

**Comment by grandchild** [comment]
@anarcat since you're "only" a member, i think moving the repo is not something you can do, i guess?

**Comment by grandchild** [comment]
If we can get people together for each operating system, who can read python and actually use selfspy that would be a great step towards keeping bugs at bay when merging things.

🗹 linux ✋🏾
❓ mac
❓ windows

**Comment by bob-ortiz** [comment]
I'm just a Mac user, but I think I could help testing.

**Comment by amerlyq** [comment]
@grandchild About "fairly active forks" in network... 
I tried to merge everything available in network. Primarily they are:

- anarcat
- davret -- Win, clipboard, Qt, etc
- giansegato -- python3, large RFC
-     + develop
- kovasap
- pablojvarela
- infokiller -- python3, large RFC, deps
- leogrin -- osx, add typespeed/quality
- lukas1994 -- osx, noWin, clipboard, chrome urls
- ozgurgundogan
- titan-c
- xamogast
- antiDigest -- push-forced
- hiromu -- osx, fix_localized_app_name
- jamesguthrie  -- osx, Prepend current URL to window name in Chrome

Result -- 6h of work down the drain with no light in the tunnel.

Basically there are 5x diff for ~2000 lines (with context), fairly similar, but fairly different and trying to disambiguate them is a hell.
If selfspy will get some maintenance on track -- contacting fork authors and asking to clean up their work in preparation to merge would be the only reasonable way forward. But as no one has time for that (fork authors already solved their "pains") -- it's easier to treat those forks as non-existent.

**Comment by Qwyntex** [comment]
I hate to bump a two year old issue but I wanted to ask for an update on this. Is this project now officially unmaintained and without hope or is there something going on behind the scenes?
As for alternatives, what did you settle with or do you keep using this project even though it is most likely outdated by this point.

**Comment by perdakovich** [comment]
Pretty much what @Qwyntex said.

---

I've been thinking about something like that for a long time. Stumbled across this program, saw the name and description, it's literally what I was thinking!
> selfspy
> Log everything you do on the computer, for statistics, future reference and all-around fun! 

It felt good. Does anyone else know this feeling? Well, whatever. 
Then I even saw the AUR package. I was very happy.

Then I saw the date of the last commitment. lmao :smiling_face_with_tear: 

---

I love the work @amerlyq has done, and also this issue from @anarcat: #160
Really cool, you can see selfspy has a lot of potential!

I'm an Arch Linux user btw, so if you need any help testing, I'm here for you!

**Comment by amerlyq** [comment]
I had given up on using selfspy that time.
Basically I rewrote only necessary part of monitoring keypresses into
python and xlib and never used even that after that, as its purpose was
fleeting.

Hooks into nvim,zsh,ranger,qtile,copyq to store full history of events in
separate git repo as text files were enough for my purposes of tracking and
searching context when what I did.

**Comment by perdakovich** [comment]
@amerlyq can you share these hooks?

**Comment by amerlyq** [comment]
@Goosegit11 Sure, you can try to adapt my mess at these permalinks to your needs:

* nvim_save: https://github.com/amerlyq/airy/blob/eccae8139f12d671e721ee4349d58cc113ec77c3/nvim/plugin/audit-events.vim
* wm_focus: https://github.com/amerlyq/airy/blob/f435b88c809651dece2fd2bc1d19aefa258b3b3b/qtile/config.py#L395
* clipboard: https://github.com/amerlyq/airy/blob/f435b88c809651dece2fd2bc1d19aefa258b3b3b/qtile/config.py#L418
* zsh_hist: https://github.com/amerlyq/airy/blob/f435b88c809651dece2fd2bc1d19aefa258b3b3b/zsh/cfg/zshrc#L56
